overlord/snapstate/handlers.go

```diff
@@ -218,10 +226,8 @@ func (m *SnapManager) doPrerequisites(t *state.Task, _ *tomb.Tomb) error {
 
 func (m *SnapManager) installOneBaseOrRequired(st *state.State, snapName, channel string, onInFlight error, userID int) (*state.TaskSet, error) {
 	// The core snap provides everything we need for core16.
-	if snapName == "core16" {
-		if hasCore, err := isInstalled(st, "core"); hasCore && err == nil {
-			return nil, nil
-		}
+	if snapName == "core16" && coreSnapInstalled(st) {
+		return nil, nil
```

Comment:
Do we have a unit test for this case?

Reply:
Yes, prereqSuite.TestDoPrereqCore16wCoreNothingToDo checks this